dlup/data/experimental/dataset.py

```diff
@@ -0,0 +1,135 @@
+# coding=utf-8
+# Copyright (c) dlup contributors
+"""Experimental dataset functions, might e.g. lack tests, or requires input from users"""
+
+import pathlib
+from typing import Callable, Iterable, List, Optional, Sequence, Tuple
+
+import numpy as np
+
+from dlup import SlideImage
+from dlup.data.dataset import TiledROIsSlideImageDataset
+from dlup.tiling import Grid, TilingMode
+
+
+class MultiScaleTiledROIsSlideImageDataset(TiledROIsSlideImageDataset):
```

Comment:
This naming is getting a bit out of hand. I think also TiledROIsSlideImageDataset it's awful. Naming is hard but it impacts a lot the final quality. Maybe name it simply MultiScaleDataset?